src/core_modules/capture-core/components/DataEntries/common/TEIAndEnrollment/useMetadataForRegistrationForm/hooks/useEnrollmentFormFoundation.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import { useIndexedDBQuery } from '../../../../../../utils/reactQueryHelpers';
+import { buildEnrollmentForm } from '../buildFunctions/buildEnrollmentForm';
+import type { TrackedEntityType, Program } from '../../../../../../metaData';
+import { scopeTypes } from '../../../../../../metaData';
+import type { CachedOptionSet, CachedTrackedEntityAttribute } from '../../../../../../storageControllers/cache.types';
+import type { DataEntryFormConfig } from '../types';
+
+type Props = {
+    scopeType: $Values<typeof scopeTypes>,
+    optionSets: ?Array<CachedOptionSet>,
+    trackedEntityType: ?TrackedEntityType,
+    trackedEntityTypeCollection: ?TrackedEntityType,
+    program: ?Program,
+    cachedTrackedEntityAttributes: ?Array<CachedTrackedEntityAttribute>,
+    dataEntryFormConfig: ?DataEntryFormConfig,
+    configIsFetched: boolean,
+    locale: string,
+};
+
+export const useEnrollmentFormFoundation = ({
+    optionSets,
+    scopeType,
+    trackedEntityType,
+    trackedEntityTypeCollection,
+    cachedTrackedEntityAttributes,
+    dataEntryFormConfig,
+    configIsFetched,
+    program,
+    locale,
+}: Props) => {
+    const { data: enrollment, isLoading, error } = useIndexedDBQuery(
+        ['enrollmentForm', program?.id],
+        () => buildEnrollmentForm({
+            // $FlowFixMe - Flow does not understand that the values are not null here
```

Comment:
Hey, there are quite a few ` // $FlowFixMe` in this file. Can anything be done to improve this?